pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -245,6 +261,24 @@ func WithLogger(logger logger.Logger) ReverseExpandQueryOption {
 	}
 }
 
+// shallowClone creates an identical copy of reverseExpandQuery except
+// candidateObjectsMap as list object candidates need to be validated
+// via check.
+func (c *ReverseExpandQuery) shallowClone() *ReverseExpandQuery {
+	return &ReverseExpandQuery{
+		logger:                  c.logger,
+		datastore:               c.datastore,
+		typesystem:              c.typesystem,
+		resolveNodeLimit:        c.resolveNodeLimit,
+		resolveNodeBreadthLimit: c.resolveNodeBreadthLimit,
+		dispatchThrottlerConfig: c.dispatchThrottlerConfig,
+		candidateObjectsMap:     new(sync.Map),
+		visitedUsersetsMap:      c.visitedUsersetsMap,
+		localCheckResolver:      c.localCheckResolver,
+		optimizationsEnabled:    c.optimizationsEnabled,
+	}
+}
```

Comment:
This could be a bit less verbose:
```suggestion
func (c *ReverseExpandQuery) shallowClone() *ReverseExpandQuery {
    copy := *c
    copy.candidateObjectsMap = new(sync.Map)
    return &copy
}
```